app/models/manageiq/providers/vmware/infra_manager/refresh_parser.rb

```diff
@@ -1029,12 +1030,13 @@ def self.folder_inv_to_hashes(inv, is_dc)
           mor = data['MOR'] # Use the MOR directly from the data since the mor as a key may be corrupt
 
           child_mors = if is_dc
-                         get_mors(data, 'hostFolder') + get_mors(data, 'vmFolder')
+                         get_mors(data, 'hostFolder') + get_mors(data, 'vmFolder') + get_mors(data, 'datastoreFolder')
                        else
                          get_mors(data, 'childEntity')
                        end
 
           new_result = {
+            :type          => type,
```

Comment:
Am I reading this right?  The type is the EmsFolder.name/Datacenter.name/StorageCluster.name?